reddit_service_activity/counter.py

```diff
@@ -43,8 +41,14 @@ def __init__(self, activity_window):
         self.slice_count = int(slice_count)
 
     def record_activity(self, redis, context_id, visitor_id):
-        key = _make_key(context_id)
-        redis.execute_command("PFADD", key, visitor_id)
+        current_slice = _current_slice()
+        key = _make_key(context_id, current_slice)
+        expiration = ((current_slice + 1) * _SLICE_LENGTH) + 1
+
+        with redis.pipeline("record") as pipe:
+            pipe.execute_command("PFADD", key, visitor_id)
+            pipe.expireat(key, expiration)
+            pipe.execute()
```

Comment:
So previously we were just waiting until we hit some limit and redis automatically evicted, causing long pauses?